plutus-tx-plugin/src/Language/PlutusTx/Compiler/Expr.hs

```diff
@@ -395,9 +395,9 @@ compileExpr e = withContextM 2 (sdToTxt $ "Compiling expr:" GHC.<+> GHC.ppr e) $
             -- See Note [Non-strict let-bindings]
             let nonStrict = not $ PIR.isTermValue arg'
             withVarScoped b $ \v -> do
-                let binds = [ PIR.TermBind () (if nonStrict then PIR.NonStrict else PIR.Strict) v arg' ]
+                let binds = pure $ PIR.TermBind () (if nonStrict then PIR.NonStrict else PIR.Strict) v arg'
                 body' <- compileExpr body
-                pure $ PIR.Let () PIR.NonRec binds body'
+                pure $ PIR.mkLet () PIR.NonRec binds body'
```

Comment:
This is fine, but the old version was still correct and more straightforward. Since you've just constructed the binds and know it's non-empty.